Features/NotesFeature/NotesViewModel.swift

```diff
@@ -66,10 +71,40 @@ final class NotesViewModel: ObservableObject {
         }
     }
 
+    func prepareNotesForSharing() async -> String {
+        do {
+            return try await crasher.recordError("Failed to share notes") {
+                var notesText = ""
+                for note in await notes {
+                    let noteText = if let noteContent = note.note.note, noteContent != "" {
+                        "Note: \(noteContent.trimmingCharacters(in: .newlines))\n"
+                    } else {
+                        ""
+                    }
+                    
+                    let verses = try await textRetriever.textForVerses(Array(note.note.verses))
+                        .filter { !$0.isEmpty }
+                        .map { $0.hasSuffix("﴾") ? String($0.dropLast()) : $0 }
```

Comment:
Do you mind explaining this logic? Just want to see if there's a problem in the textForVerses method.

Reply:
The arabic text from `textForVerses` has a trailing bracket that looks odd so I opted to drop it from the string.

It comes from [here](https://github.com/quran/quran-ios/blob/4df3eb4a8c91571c18d9e8e38592bf7a0bad6b50/Domain/QuranTextKit/Sources/ShareableText/ShareableVerseTextRetriever.swift#L66-L69) I believe. You can also see some more examples of the output [here](https://github.com/quran/quran-ios/blob/4df3eb4a8c91571c18d9e8e38592bf7a0bad6b50/Domain/QuranTextKit/Tests/ShareableVerseTextRetrieverTests.swift#L59-L63).

I'm not sure if it's a bug or a result of some other logic since the opening bracket at the end matches the closing bracket at the next ayah.

Reply:
<img width="387" alt="image" src="https://github.com/user-attachments/assets/57e1841c-c8c8-4ae0-9520-954c82c063ba">

Reply:
Oh, I see. This is actually an issue with renderer when it comes to combining Arabic and English letters. The brackets `()` are part of English (left-to-right) while the text is arabic (right-to-left) and end of the ayah is an arabic number surrounded by the brackets (e.g. `(1)`). So most renderer just show them broken and show the closing bracket at the start of the sentence from right. So what we could do instead is to replace `(1)`, `(2)`, ... these with just some other separator that would render right (e.g. `۞`). What do you think?

Reply:
I'm going to approve the PR, but feel free to fix this as a follow up PR or I can fix it myself this weekend.

Reply:
Thanks! I opened up a PR with a bit of a hacky solution but it might work. It's also not tested on anything besides English & Arabic, but maybe this could be a start.

https://github.com/quran/quran-ios/pull/650